weatherman.py

```diff
@@ -0,0 +1,178 @@
+import argparse
+import calendar
+from datetime import datetime
+import os
+
+RED = "\033[31m"
+BLUE = "\033[34m"
+RESET = "\033[0m"
+
+
+class WeatherReading:
+    def __init__(self, observation_date, max_temp, min_temp, max_humidity, mean_humidity):
+        self.date = observation_date
+        self.max_temp = max_temp
+        self.min_temp = min_temp
+        self.max_humidity = max_humidity
+        self.mean_humidity = mean_humidity
+                                                                                                   
+    def __repr__(self):
+        return (f"WeatherReading(date={self.date}, max_temp={self.max_temp}, min_temp={self.min_temp},"
+                f"max_humidity={self.max_humidity}, mean_humidity={self.mean_humidity})")
+
+
+def parse_weather_file(file_path):
+    readings = []
+    try:
+        with open(file_path, 'r') as file:
+            header = [item.strip() for item in next(file).strip().split(',')]
+            required_fields = [
+                "PKT", "Max TemperatureC", "Min TemperatureC", "Max Humidity", "Mean Humidity"]
+            field_indices = {field: header.index(field) for field in required_fields}
+
+            for line in file:
```

Comment:
Which `line` and `file` are we talking about? Both are generic and vague so let's update them to something meaningful.